static/js/lightbox.js

```diff
@@ -210,14 +210,23 @@ function display_image(payload) {
     img.src = payload.source;
     $img_container.html(img).show();
 
-    const filename = payload.url?.split("/").pop();
+    const url = new URL(payload.url, window.location.href);
+    const filename = url.pathname.split("/").pop();
     $(".image-description .title")
         .text(payload.title || "N/A")
         .attr("aria-label", payload.title || "N/A")
         .prop("data-filename", filename || "N/A");
     $(".image-description .user").text(payload.user).prop("title", payload.user);
 
-    $(".image-actions .open, .image-actions .download").attr("href", payload.source);
+    $(".image-actions .open").attr("href", payload.source);
+    const same_origin = url.origin == window.location.origin;
+    if (same_origin && url.pathname.startsWith("/user_uploads/")) {
+        // Switch to the "download" handler, so S3 URLs set their Content-Disposition
+        url.pathname = url.pathname.replace("/user_uploads/", "/user_uploads/download/");
```

Comment:
`"/user_uploads/download/" + url.pathname.slice("/user_uploads/".length)` more clearly expresses that we’re adjusting the beginning and not somewhere in the middle.

Reply:
I feel like the `.startsWith` directly above that makes it fairly clear, but I'm fine with your suggestion as well.